src/main/java/dev/galacticraft/mod/client/gui/screen/ingame/AirlockControllerScreen.java

```diff
@@ -22,46 +22,328 @@
+                publicBtn.setSelected(now == ProximityAccess.PUBLIC);
+                teamBtn.setSelected(now == ProximityAccess.TEAM);
+                privateBtn.setSelected(now == ProximityAccess.PRIVATE);
+            }
+            this.cachedAccess = now;
+        }
+    }
+
+    public AirlockControllerScreen(AirlockControllerMenu menu, Inventory inv, Component title) {
+        super(menu, title, Constant.ScreenTexture.AIRLOCK_CONTROLLER_SCREEN);
+
+        this.textField = new EditBox(
+                Minecraft.getInstance().font,
+                this.leftPos + 132, this.topPos + 65,
+                26, 20,
+                Component.empty()
+        );
+        this.textField.setValue(String.valueOf(this.menu.proximityOpen));
+
+        this.textField.setFilter(s -> {
+            if (s.isEmpty()) return true;
+            try {
+                int v = Integer.parseInt(s);
+                return v >= 0 && v <= 5;
+            } catch (NumberFormatException ignore) {
+                return false;
+            }
+        });
+
+        this.textField.setResponder(s -> {
+            if (s.isEmpty()) return;
+            byte value;
+            try {
+                value = Byte.parseByte(s);
+            } catch (NumberFormatException ignore) {
+                return;
+            }
+            if (value < 0) value = 0;
+            if (value > 5) value = 5;
+
+            String clamped = String.valueOf(value);
+            if (!s.equals(clamped)) {
+                this.textField.setValue(clamped);
+                return;
+            }
+
+            if (value != this.menu.proximityOpen) {
+                this.menu.proximityOpen = value;
+                ClientPlayNetworking.send(new AirlockSetProximityPayload(value));
+            }
+        });
     }
 
     @Override
     protected void init() {
         super.init();
-        addRenderableWidget(new CheckboxButton(this.leftPos + 8, this.topPos + 20));
+        this.imageHeight = 171;
+        this.titleLabelX = 90;
+
+        this.textField.setX(this.leftPos + 132);
+        this.textField.setY(this.topPos + 65);
+        this.addRenderableWidget(this.textField);
+
+        ProximityAccess initial = currentAccess();
+        switch (initial) {
+            case PUBLIC  -> selected = AccessMode.PUBLIC;
+            case TEAM    -> selected = AccessMode.TEAM;
+            case PRIVATE -> selected = AccessMode.PRIVATE;
+        }
+
+        this.publicBtn = new IconButton(0, 0, ACCESS_BTN_SIZE, MACHINELIB_PANELS, PUB_U,  PUB_V,  PUB_W,  PUB_H, b -> {
+            selected = AccessMode.PUBLIC;
+            publicBtn.setSelected(true); teamBtn.setSelected(false); privateBtn.setSelected(false);
+            this.menu.proximityAccess = ProximityAccess.PUBLIC;
+            ClientPlayNetworking.send(new AirlockSetProximityAccessPayload(ProximityAccess.PUBLIC));
+        }, Component.literal("Public"));
```

Comment:
This should really be translatable. You could reuse MachineLib's translations for the access levels.

```suggestion
        }, AccessLevel.PUBLIC.getName());
```